src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -42,156 +43,31 @@ const HierarchicalTree = (props) => {
-            }
-          }
-
-          if (element.children) {
-            addToChildren(element.children, key);
-          }
-        });
-      };
-
-      addToChildren(newTreeData, dropKey);
-    }
+    // If rootNode, ignore
+    if (dragNode.rootNode) return;
 
-    // We dropped the item into a gap between two items.
-    if (info.dropToGap) {
-      // It can either be above or below an element.
-      const dropPosition = info.node.dragOverGapTop ? 'top' : 'bottom';
-
-      const addIntoGap = (haystack, needle) => transform(
-        haystack,
-        (result, objValue) => {
-          // We found the place to insert the dragged element.
-          if (objValue.key === needle) {
-            /**
-               * If a dragged object is a root node, it should only be
-               * droppable in between root nodes.
-               */
-            if (!objValue.rootNode && dragObj.rootNode === true) {
-              shouldUpdateState = false;
-              return false;
-            }
-
-            /**
-               * If a dragged object is not a root node, it should only be
-               * droppable in between non-root nodes.
-               */
-            if (!dragObj.rootNode && objValue.rootNode === true) {
-              shouldUpdateState = false;
-              return false;
-            }
-
-            if (objValue.children) {
-              // eslint-disable-next-line no-param-reassign
-              objValue.children = addIntoGap(objValue.children, needle);
-            }
-
-            /**
-               * Add the drag object into the tree according to where it was
-               * dropped.
-               */
-            if (dropPosition === 'top') {
-              result.push(dragObj);
-              result.push(objValue);
-            } else {
-              result.push(objValue);
-              result.push(dragObj);
-            }
-          } else {
-            /**
-               * We have not found the place to insert the element.
-               * Carry on building the new object until we are there.
-               */
-            if (objValue.children) {
-              // eslint-disable-next-line no-param-reassign
-              objValue.children = addIntoGap(objValue.children, needle);
-            }
-            result.push(objValue);
-          }
-
-          return true;
-        },
-        [],
-      );
+    // pos is a string e.g.: 0-0-1, each number is a position in a tree level
+    const posFromArray = dragNode.pos.split('-');
+    const posToArray = node.pos.split('-');
 
-      newTreeData = addIntoGap(newTreeData, dropKey);
-    }
+    const fromPosition = parseInt(posFromArray[2], 10);
 
-    if (shouldUpdateState) {
-      props.onHierarchyUpdate(newTreeData);
-    }
+    // If not in the same cellClass, ignore
+    if (!_.isEqual(posFromArray[1], posToArray[1])) return;
+
+    // If was dropped in same place, ignore
+    if (fromPosition === dropPosition) return;
```

Comment:
It is the index on which the item is dropped and the item should be inserted into.

However, it changes to a different value if the mouse is dragged more towards the right, so when it is inserted into an item on its same level, like for example if we wanted to insert a cellSet into another cellSet (although in this case we get a `dropToGap === true` so that's the way we can deal with that case).

There is only one case in which `dropToGap === true` is fine, and that is when the node we are inserting our item into is the rootNode, and this case is only really relevant to us when the item is inserted at `dropPosition === 0`.I know it is kind of confusing but it's the way `Tree` was coded in antd, and I'm sure it makes more sense if you allow multiple levels on the `Tree` (instead of only 2 like we need).

Reply:
If you want us to go over this in more detail we can have a call tomorrow, let me know

Reply:
Can you add this information in a comment?